lib/msf/ui/debug.rb

```diff
@@ -0,0 +1,258 @@
+# -*- coding: binary -*-
+# frozen_string_literal: true
+
+module Msf
+  module Ui
+    ###
+    #
+    # Displays Metasploit information useful for Debugging.
+    #
+    ###
+    module Debug
+      COMMAND_HISTORY_TOTAL = 50
+      ERROR_TOTAL = 10
+      LOG_LINE_TOTAL = 50
+      ISSUE_LINK = 'https://github.com/rapid7/metasploit-framework/issues/new'
+      PREAMBLE = <<~PREMABLE
+          Please provide the below information in any Github issues you open. New issues can be opened here #{ISSUE_LINK.dup}
+          %red%undENSURE YOU HAVE REMOVED ANY SENSITIVE INFORMATION BEFORE SUBMITTING!%clr
+
+          ===8<=== CUT AND PASTE EVERYTHING BELOW THIS LINE ===8<===
+
+
+      PREMABLE
+
+
+
+      def self.issue_link
+        return ISSUE_LINK.dup
+      end
+
+      def self.preamble
+        return PREAMBLE.dup
+      end
+
+      def self.all(framework, driver)
+        all_information = preamble
+        all_information << datastore(framework, driver)
+        all_information << history
+        all_information << errors
+        all_information << logs
+        all_information << versions(framework)
+
+        all_information
+      end
+
+      def self.datastore(framework, driver)
+        # Generate an ini with the existing config file
+        ini = Rex::Parser::Ini.new(Msf::Config.config_file)
+
+        # Delete all groups from the config ini that potentially have more up to date information
+        ini.keys.each do |k|
+          unless k =~ %r{^framework/database}
+            ini.delete(k)
+          end
+        end
+
+        # Retrieve and add more up to date information
+        add_hash_to_ini_group(ini, framework.datastore, driver.get_config_core)
+        add_hash_to_ini_group(ini, driver.get_config, driver.get_config_group)
+
+        if driver.active_module
+          ds = driver.active_module.datastore.dup
+
+          # Ensures that the local default value of a variable isn't set if a global value of a variable is set
+          ds.keys.each do |k|
+            # Removes an active module datastore item if it has a default value & it matches the default value
+            if driver.active_module.options.key?(k.upcase) && driver.active_module.options[k.upcase].default == ds[k]
```

Comment:
Been a while since I wrote this, so will poke at it and double check, but I do agree with you on bad defaults. 

IIRC the defaults _could_ be pretty numerous and my thinking was it was better to prioritize the User set values standing out, hadn't considered defaults being the issue or becoming the issue if users are running older builds.

Reply:
Double checked and the difference in size with the above code commented out is negligible. I'll include default values, and mark them as default to aid debugging